ReactWindows/ReactNative.Shared/ReactInstanceManager.cs

```diff
@@ -411,19 +411,21 @@ public void OnResume(Action onBackPressed)
 
             DispatcherHelpers.AssertOnDispatcher(rootView);
 
-            await DispatcherHelpers.CallOnDispatcher(() =>
+            var detachTask = await DispatcherHelpers.CallOnDispatcher(() =>
             {
                 if (_attachedRootViews.Remove(rootView))
                 {
                     var currentReactContext = _currentReactContext;
                     if (currentReactContext != null && currentReactContext.HasActiveReactInstance)
                     {
-                        DetachViewFromInstance(rootView, currentReactContext.ReactInstance);
+                        return DetachViewFromInstanceAsync(rootView, currentReactContext.ReactInstance);
                     }
                 }
 
-                return true;
+                return Task.CompletedTask;
             }, true); // inlining allowed
+
+            await detachTask;
         }
```

Comment:
I don't see how this is any different than using "Unwrap()" as you had originally done.

Reply:
Unwrap hid some logic that was kind of hard to speculate about, so I converted to a clearer/more  explicit flow.


---
In reply to: [179480881](https://github.com/Microsoft/react-native-windows/pull/1737#discussion_r179480881) [](ancestors = 179480881)